plugins/e57/io/E57Writer.cpp

```diff
@@ -81,12 +82,25 @@ void E57Writer::ChunkWriter::write(pdal::PointRef& pt)
     }
 
     // Add point to buffer and increase index
+    using DimId = pdal::Dimension::Id;
     for (auto& keyValue: m_doubleBuffers)
     {
         auto pdaldim = pdal::e57plugin::e57ToPdal(keyValue.first);
-        if (pdaldim != pdal::Dimension::Id::Unknown)
+        if (pdaldim != DimId::Unknown)
         {
             auto val = pt.getFieldAs<double>(pdaldim);
+            if ((pdaldim == DimId::Red || pdaldim == DimId::Green ||
+                    pdaldim == DimId::Blue) &&
+                    val > colorLimit)
+            {
+                colorLimit = colorLimit << 8;  // Increase color bytes.
+            }
+
+            if (pdaldim == DimId::Intensity && val > intensityLimit)
+            {
+                while (val > intensityLimit)
+                    intensityLimit *= 10;  // Intensity limits values can be one of (1,10,100,...)
```

Comment:
Why incrementing by 10?

Reply:
Assumming that intensity range will be either 0-1, 0-10, 1-100...

Reply:
Humm not sure that's a correct assumption. Much better to replace with `m_intensityLimit = max(val,m_intensityLimit)`.

Reply:
done